ldelf/ta_elf_rel.c

```diff
@@ -120,25 +129,55 @@ static TEE_Result resolve_sym_helper(uint32_t hash, const char *name,
 					  ELF64_ST_TYPE(sym[n].st_info),
 					  sym[n].st_shndx,
 					  sym[n].st_name,
-					  sym[n].st_value, name, val))
+					  sym[n].st_value, name, val, weak_ok,
+					  found_weak_undef))
 				return TEE_SUCCESS;
 		}
 	}
 
 	return TEE_ERROR_ITEM_NOT_FOUND;
 }
 
+/*
+ * Look for named symbol in @elf, or all modules if @elf == NULL. Global symbols
+ * are searched first, then weak ones. Last option, when at least one weak but
+ * undefined symbol exists, resolve to zero. Otherwise return
+ * TEE_ERROR_ITEM_NOT_FOUND.
+ */
 TEE_Result ta_elf_resolve_sym(const char *name, vaddr_t *val,
 			      struct ta_elf *elf)
 {
 	uint32_t hash = elf_hash(name);
+	bool found_weak_undef = false;
 
-	if (elf)
-		return resolve_sym_helper(hash, name, val, elf);
+	if (elf) {
+		/* Search global symbols */
+		if (!resolve_sym_helper(hash, name, val, elf, false, NULL))
```

Comment:
I woulnd't mind a `false/*!weak_ok*/` instead of just `false`.

Reply:
Sure